### PR TITLE
Remove invalid assertion

### PR DESCRIPTION
**Pull request by markuspf**
This is really only relevant in maintainer mode, in released versions the code behaves as expected.

### CODE REVIEW

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/9714/

File: arangod/Aql/PruneExpressionEvaluator.cpp [hunk 1]
```diff
@@ -45,6 +45,5 @@ bool PruneExpressionEvaluator::evaluate() {
   bool mustDestroy = false;
   aql::AqlValue res = _pruneExpression->execute(&_ctx, mustDestroy);
   arangodb::aql::AqlValueGuard guard(res, mustDestroy);
-  TRI_ASSERT(res.isBoolean());
```
Comment (mchacki) [inline arangod/Aql/PruneExpressionEvaluator.cpp:48]: Yes, we can actually cast many thing to bool, so not required to assert that we have a bool in hand

File: tests/js/server/aql/aql-traversal-wrong-assertion-regression.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Copyright 2020, triAGENS GmbH, Cologne, Germany
```
Comment (mchacki) [inline tests/js/server/aql/aql-traversal-wrong-assertion-regression.js:22]:
```suggestion
/// @author Copyright 2020, ArangoDB GmbH, Cologne, Germany
```

I feel the need to say that we founded the Arango company a while ago ;)

**Comment by markuspf** [comment]
Do not merge while I fix up the (fairly useless) regression test.

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/9717/

**Comment by jsteemann** [comment]
jslint doesn't like it:
```
2020-04-30T15:32:49Z [7] ERROR [10407] jslint(ERROR): /work/ArangoDB/tests/js/server/aql/aql-traversal-wrong-assertion-regression.js:95,13 (Rule: no-undef, Severity: 2) 'traversalResetRegressionSuite' is not defined.
```
Seems that test never ran.

**Comment by markuspf** [comment]
@jsteeman did you see the second jenkins run where I fixed the test?

**Pull request merged**